gprofiler/profilers/python.py

```diff
@@ -245,27 +246,36 @@ def _select_processes_to_profile(self) -> List[Process]:
         filtered_procs = []
         for process in pgrep_maps(DETECTED_PYTHON_PROCESSES_REGEX):
             try:
-                if process.pid == os.getpid():
-                    continue
-
-                cmdline = process.cmdline()
-                if any(item in cmdline for item in _BLACKLISTED_PYTHON_PROCS):
-                    continue
-
-                # PyPy is called pypy3 or pypy (for 2)
-                # py-spy is, of course, only for CPython, and will report a possibly not-so-nice error
-                # when invoked on pypy.
-                # I'm checking for "pypy" in the basename here. I'm not aware of libpypy being directly loaded
-                # into non-pypy processes, if we ever encounter that - we can check the maps instead
-                if os.path.basename(process_exe(process)).startswith("pypy"):
-                    continue
-
-                filtered_procs.append(process)
+                if not self._should_skip_process(process):
+                    filtered_procs.append(process)
             except Exception:
                 logger.exception(f"Couldn't add pid {process.pid} to list")
 
         return filtered_procs
 
+    def _should_profile_process(self, process: Process) -> bool:
+        return any(
+            re.match(DETECTED_PYTHON_PROCESSES_REGEX, line) for line in read_proc_file(process, "maps").splitlines()
```

Comment:
I don't think you need the line split?

Although.. to mimic `pgrep_maps`, I suppose we do. Wdyt